tests/testutils/telemetry/traces.go

```diff
@@ -0,0 +1,257 @@
+// Copyright Splunk, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package telemetry
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+
+	"gopkg.in/yaml.v2"
+)
+
+// ResourceTraces is a convenience type for test helpers and assertions.
+type ResourceTraces struct {
```

Comment:
It's a bit difficult for me not to bikeshed that this should be `ResourceSpans` given the yaml key name and existing convention (which would translate to `AssertAllSpansReceived` in the sink) but I'm ok w/ mirroring the traces envelope given the proto definitions.

Reply:
I had the same feeling but thought in following the proto definitions...